### DISCUSSION TITLE
Error resolving ref on Binder

### DISCUSSION THREAD

**Discussion by mgenet** [executablebooks/meta#1018] [General]
Hello,
I am trying to make a Jupyter Book with links to Binder, cf. https://gitlab.inria.fr/mgenet/dolfinwarp-jupyterbook
In the `_config.yml` file I have `binderhub_url: "https://mybinder.org"` (cf. https://gitlab.inria.fr/mgenet/dolfinwarp-jupyterbook/-/blob/master/_config.yml), the generated book does have the Binder button (cf. https://mgenet.gitlabpages.inria.fr/dolfinwarp-jupyterbook/tuto_warp.html), but Binder fails with the following error:
```
Error resolving ref for git:https%3A//gitlab.inria.fr/mgenet/dolfinwarp-jupyterbook/master: Unable to run git ls-remote to get the `resolved_ref`: ssh: Could not resolve hostname https: Name or service not known
fatal: Could not read from remote repository.
```
Any idea? Thanks in advance!

**Comment by rgaiacs** [comment]
Can you try https://mybinder.org/v2/git/https%3A%2F%2Fgitlab.inria.fr%2Fmgenet%2Fdolfinwarp-jupyterbook.git/master ?

**Comment by mgenet** [comment]
Thank you for your help!
This link works.
I don't understand why the Jupyter Book has the wrong address `git:https%3A//gitlab.inria.fr/mgenet/dolfinwarp-jupyterbook/master`, whereas it should be `git:https%3A%2F%2Fgitlab.inria.fr%2Fmgenet%2Fdolfinwarp-jupyterbook%2Fmaster`
Thanks again.

**Comment by mgenet** [comment]
PS: Interestingly, if I put my repo on GitHub and write `url: https://github.com/mgenet/dolfinwarp-jupyterbook` in my `_config.yml` file it works! Somehow the Jupyter Book expects a GitHub repo?!

**Comment by alicepeyraut** [comment]
Hello,
I have the exact same issue: I am trying to make a Jupyter Book with links to Binder, but the address is wrong if the repository is on gitlab: 

```
Error resolving ref for git:https%3A//gitlab.inria.fr/apeyraut/identification-paper-demos/master: Unable to run git ls-remote to get the `resolved_ref`: ssh: Could not resolve hostname https: Name or service not known
fatal: Could not read from remote repository. 

```

Even though the link https://mybinder.org/v2/git/https%3A%2F%2Fgitlab.inria.fr%2Fapeyraut%2Fidentification-paper-demos.git/master) works.
Moreover if I make the Jupyter notebooks with the repository on Github, everything works fine. 
Is there an update on why the Jupyter Notebooks have the wrong address when the repository is on Gitlab?
Thanks in advance!

**Comment by pgierz** [comment]
I have the same problem with our own gitlab server. How can this be fixed?

**Reply by mgenet** [reply]
Never figured this out…workaround is to mirror the repo on GitHub, and use the GitHub repo for the Binder link…

**Reply by pgierz** [reply]
That's not really a solution. What if I do not want or am not able to use GitHub for whatever reason?

**Reply by mgenet** [reply]
I agree. That's why I said "workaround". Please let me know if you find an actual solution.
